src/utils/split-heads-from-rest.js

```diff
@@ -0,0 +1,25 @@
+/** Created on 8/22/16. */
+
+export default function splitHeadsFromRest(nodes = [], links = []) {
+  const heads = nodes.filter(
+    ({props: {name}}) => {
+      return !links.filter(({props: {to}}) => (to === name)).length;
+    }
+  );
+
+  const linksFrom = links.filter(
+    ({props: {from}}) => {
+      return heads.filter(({props: {name}}) => (from === name)).length;
+    }
+  );
```

Comment:
I updated the filter to find so that the inner iteration stops as soon as it finds a hit.

Also it turned out that these are iterating on two different arrays, so they can not be done in a single pass.